main.go

```diff
@@ -124,8 +127,8 @@ func main() {
 
 	submSpec.CableDriver = strings.ToLower(submSpec.CableDriver)
 
-	localEndpoint, err := util.GetLocalEndpoint(submSpec.ClusterID, submSpec.CableDriver, nil, submSpec.NatEnabled,
-		localSubnets, util.GetLocalIP(), submSpec.ClusterCidr)
+	localEndpoint, err := util.GetLocalEndpoint(submSpec.ClusterID, submSpec.CableDriver, nil,
```

Comment:
GetLocalEndpoint starts to have a lot of the submSpec details (even globalnetEnabled is actually derived from submSpec.). 

I propose we pass the submSpec down to GetLocalEndpoint.